yaksh/static/yaksh/js/requesthandler.js

```diff
@@ -0,0 +1,139 @@
+        request_status = "initial";
+        count = 0;
+        notify("Request timeout. Try again later");
+        clearInterval(checker);
+        unlock_screen();
+    } else {
+        request_status = "initial";
+        count = 0;
+        notify("Please try after few minutes");
+        clearInterval(checker);
+        unlock_screen();
+    }
+}
+
+function notify(text) {
+    var $notice = document.getElementById("notification");
+    $notice.classList.add("alert");
+    $notice.classList.add("alert-success");
+    $notice.innerHTML = text;
+}
+
+function lock_screen() {
+    document.getElementById("ontop").style.display = "block";
+}
+
+function unlock_screen() {
+    document.getElementById("ontop").style.display = "none";
+}
+
+function check_lock_screen() {
+    var $ontop_div = document.getElementById("ontop");
+    if ($ontop_div.style.display == "block") {
+        $ontop_div.style.display = "none";
+    }
+}
+
+function get_result(uid){
+    var url = "/exam/get_results/"+uid+"/";
+    ajax_call(url, "GET", "html", null, uid)
+}
+
+function ajax_call(url, method_type, data_type, data, uid) {
+    $.ajax({
+        method: method_type,
+        url: url,
+        data: data,
+        dataType: data_type, // Your server can response html, json, xml format.
+        success: function(data, status, xhr) {
+            content_type = xhr.getResponseHeader("content-type");
+            if(content_type.indexOf("text/html") !== -1) {
+                if( method_type === "POST") {
+                    request_status = "initial";
+                    count = 0;
+                }
+                clearInterval(checker);
+                unlock_screen();
+                document.open();
+                document.write(data);
+                document.close();
+            } else if(content_type.indexOf("application/json") !== -1) {
+                res = JSON.parse(data);
+                request_status = res.status;
+                if(method_type === "POST") {
+                    uid = res.uid;
+                }
+                check_state(request_status, uid);
+            } else {
+                request_status = "initial";
+                count = 0;
+                clearInterval(checker);
+                unlock_screen();
+            }
+        },
+        error: function(xhr, text_status, error_thrown ) {
+            request_status = "initial";
+            count = 0;
+            clearInterval(checker);
+            unlock_screen();
+            notify("There is some problem. Try later.")
+        }
+    });
+
+}
+
+var global_editor = {};
+$(document).ready(function(){
+  // Codemirror object, language modes and initial content
+  // Get the textarea node
+  var textarea_node = document.querySelector('#answer');
+
+  var mode_dict = {
+    'python': 'python',
+    'c': 'text/x-csrc',
+    'cpp': 'text/x-c++src',
+    'java': 'text/x-java',
+    'bash': 'text/x-sh',
+    'scilab': 'text/x-csrc'
```

Comment:
I don't like this hard-coded list of languages that probably needs to be changed in multiple places  but that is for another PR...

Reply:
Created an issue